### DISCUSSION TITLE
Pioasm + Platorfmio

### DISCUSSION THREAD

**Discussion by hexeguitar** [earlephilhower/arduino-pico#998] [Ideas]
In case you find pasting the pio assembler code to the online compiler too annoying here is a method to automate it in Platformio projects:  
https://gist.github.com/hexeguitar/f4533bc697c956ac1245b6843e2ef438

It uses a python script invoked before the build starts which scans the *src* directory for *.pio* files and converts them using installed pioasm into h include files.  

Hope some of you will find it useful!
Piotr

**Comment by earlephilhower** [comment]
Do you think this is something you should add to the Platform.IO scripts at https://github.com/platformio/platform-raspberrypi/pull/36 ?

**Reply by djessup** [reply]
Yes please!

**Comment by hexeguitar** [comment]
Since the earlephilhower's pioasm tool has been removed from the Platformio repository and it's not installed by default, the previous version stopped to work. Updated the gist to manually install the tool via a direct link to the package.
